lib/AdminPages/Pages/Segments.php

```diff
@@ -32,6 +49,15 @@ public function render() {
     $data['has_valid_api_key'] = $this->subscribersFeature->hasValidApiKey();
     $data['subscriber_count'] = Subscriber::getTotalSubscribers();
 
+    $data['mss_key_invalid'] = false;
+
+    if ($this->servicesChecker->isMailPoetAPIKeyValid() === false) {
+      $mssKey = $this->settings->get(Bridge::API_KEY_STATE_SETTING_NAME);
+      if (isset($mssKey['state']) && ($mssKey['state'] === Bridge::KEY_INVALID)) {
+        $data['mss_key_invalid'] = true;
+      }
```

Comment:
I see these lines repeated in 2 places: here and in SubscribersPage. What about adding a method to servicesChecker ?